Core/include/Acts/TrackFitting/KalmanFitter.hpp

```diff
@@ -477,6 +490,24 @@ class KalmanFitter {
         ACTS_VERBOSE("Measurement surface " << surface->geometryId()
                                             << " detected.");
 
+        // If holes and material interaction occured since
+        // the last measurement, store them in the MultiTrajectory.
+        if (result.endTip != SIZE_MAX) {
+          for (size_t i = 0; i <= result.endTip; i++) {
+            auto endState = result.endNonMeasurementStates.getTrackState(i);
+            result.trackTip = result.fittedStates.addTrackState(
+                endState.getMask(), result.trackTip);
+            // Copy the hole/material state in the MultiTrajectory.
+            result.fittedStates.getTrackState(result.trackTip)
+                .copyFrom(endState);
+            // We count the processed state
+            ++result.processedStates;
+          }
```

Comment:
I see what this does. I'm honestly not sure if this is not actually slower than just plugging the holes into the normal storage. This also additionally incurs a copy operation.

I think putting them in the end and then discarding them during trajectory extraction would be the way to go.

Reply:
As I said during the meeting I fully agree. I just thought of one thing so let me know what you think about it : instead of having one trackTip in the KalmanFitterResult we could have two, one that correspond to the last measurement and one that correspond to the end of the Multitrjectory (after the extra holes and material interaction). That way we avoid rerunning over those extra state in the smoothing and it is easier to access the "interesting" measurement+ holes.

Reply:
Yeah, that could work. Then it's up to the collector which track tip to start from, and if they are identical, they know what the situation is as well.